docs/cmd/changelogutils/changelog_utils.go

```
@@ -0,0 +1,288 @@
+	for n := rootNode.FirstChild(); n != nil; n = n.NextSibling() {
+		switch typedNode := n.(type) {
+		case *ast.Paragraph:
+			{
+				if child := typedNode.FirstChild(); child.Kind() == ast.KindEmphasis {
+					emphasis := child.(*ast.Emphasis)
+					if emphasis.Level == 2 {
+						// Header block
+						currentHeader = string(typedNode.Text(releaseNotesBuf))
+					}
+
+				} else {
+					v := typedNode.Lines().At(0)
+					note := prefix + fmt.Sprintf("%s\n", v.Value(releaseNotesBuf))
+					if currentHeader != "" {
+						//any extra text e.g. "This release build has failed", only used for enterprise release notes
+						headersToNotesMap[currentHeader] = append(headersToNotesMap[currentHeader], note)
+					} else {
+						accumulator = accumulator + note
+					}
+				}
+			}
+		case *ast.List:
+			{
+				// Only add release notes if we are under a current header
+				if currentHeader != "" {
+					for child := n.FirstChild(); child != nil; child = child.NextSibling() {
+						v := child.FirstChild().Lines().At(0)
+						releaseNoteNode := v.Value(releaseNotesBuf)
+						releaseNote := prefix + string(releaseNoteNode)
+						headersToNotesMap[currentHeader] = append(headersToNotesMap[currentHeader], releaseNote)
+					}
+				}
+			}
+		}
+	}
+	return accumulator
+}
+
+func GetOSDependencyPrefix(openSourceVersion Version, isHeader bool) string {
+	prefix := "\n- (From"
+	if isHeader {
+		prefix = " (Uses Gloo Edge"
+	}
+	osReleaseURL := strings.ReplaceAll(openSourceVersion.String(), ".", "")
+	osPrefix := fmt.Sprintf("%s [OSS %s](/reference/changelog/open_source/#%s)) ", prefix, openSourceVersion.String(), osReleaseURL)
+	return osPrefix
+}
+
+func GetPreviousEnterprisePatchVersion(enterpriseReleasesSorted []*github.RepositoryRelease, index int) *Version {
+	var previousEnterpriseVersion *Version
+	currentVersion, _ := ParseVersion(enterpriseReleasesSorted[index].GetTagName())
```

why are we swallowing errors?